### PR TITLE
Feature/mini app web simulator

### PR DESCRIPTION
**Pull request by ZijieZh**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "cpx": "^1.5.0"
   },
   "dependencies": {
+    "@ali/minidev-rc": "^1.2.11-game.2",
```
Comment (lou1swu) [inline packages/cli/package.json]: 内部依赖换成外部依赖

File: packages/cli/src/actions/openMiniAppSimulator.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  // write port file
+  writePortFile(proxyServerPort);
+  logger.success('port file written');
+  const app = express();
+  app.listen(assetsFinalPort);
+  app.use('/', express.static(__dirname));
+
+  // 启动minidev
+  const { minidev, } = require('@ali/minidev-rc');
```
Comment (lou1swu) [inline packages/cli/src/actions/openMiniAppSimulator.ts]: 内部依赖问题

File: packages/cli/src/actions/openMiniAppSimulator.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    monitor.logJSError(e);
+    return;
+  }
+
+  // write port file
+  writePortFile(proxyServerPort);
+  logger.success('port file written');
+  const app = express();
+  app.listen(assetsFinalPort);
+  app.use('/', express.static(__dirname));
```
Comment (lou1swu) [inline packages/cli/src/actions/openMiniAppSimulator.ts]: express 换成http-server吧，这样就不用引入express这个依赖了

File: packages/cli/src/actions/openMiniAppSimulator.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  try {
+    assetsFinalPort = await choosePort(DEFAULT_HOST, assetsPort);
+  } catch(e) {
+    logger.error('assetsPort choose fail', e);
+    monitor.logJSError(e);
+    return;
+  }
+
+  // write port file
+  writePortFile(proxyServerPort);
```
Comment (lou1swu) [inline packages/cli/src/actions/openMiniAppSimulator.ts]: 这里写失败时，catch了错误。下面显示执行成功是无效的

File: packages/cli/package.json (hunk 1 above)
Comment (ZijieZh) [reply packages/cli/package.json]: done

File: packages/cli/src/actions/openMiniAppSimulator.ts (hunk 2 above)
Comment (ZijieZh) [reply packages/cli/src/actions/openMiniAppSimulator.ts]: done

File: packages/cli/src/lib/util/writePortFile.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import fs from 'fs';
+import path from 'path';
+import { logger, } from '../cli-shared-utils/lib/logger';
+
+export default function writePortFile(port: number): void {
+  try {
+    fs.writeFileSync(path.join(__dirname, '../../actions/port.js'), `window.miniapp_proxy_server_port = ${port};`);
+  } catch (e) {
+    logger.error('port.js write fail', e);
+    return;
```
Comment (lou1swu) [inline packages/cli/src/lib/util/writePortFile.ts]: 这里是隐患，这里应该把错误抛出，外层调用时再做catch出来，否则外层调用时，无论文件写入成功或失败，逻辑都会继续执行。

**Review by lou1swu — APPROVED** [review]

**Pull request merged**